projects/FHE-DKSAP.md

```diff
@@ -0,0 +1,129 @@
+- Public key and secret key generations: 
+- ECDH algorithm
+- Stealth private key derivation
+
+Public key and secret key generations:
+
+PK_scan = G * sk_scan, PK_spend = G * sk_spend, R = G * s 
+
+All the private keys only have one part, and the G contains (x,y) . Thus, all the public keys are with (x,y) two parts.
+
+The output is similar like this:
+
+Alice’s private key: 0xe32868331fa8ef0138de0de85478346aec5e3912b6029ae71691c384237a3eeb
+
+Alice’s public key: (0x86b1aa5120f079594348c67647679e7ac4c365b2c01330db782b0ba611c1d677, 0x5f4376a23eed633657a90f385ba21068ed7e29859a7fab09e953cc5b3e89beba)
+
+- ECDH algorithm: ss = r * PK_scan = R * sk_scan
+
+sk_scan and r are secret keys with only one part. R and PK_scan are public keys with two parts, thus the shared secret key is two parts as well. For example: 
+Shared secret: (0x3e2ffbc3aa8a2836c1689e55cd169ba638b58a3a18803fcf7de153525b28c3cd, 0x43ca148c92af58ebdb525542488a4fe6397809200fe8c61b41a105449507083)
+
+- Key blinding algorithm:
+
+public key of the address: S_sa = h(ss) * G + P_spend
+
+h(ss) uses keccak-256 and the output is 256 bits as well as our private key. Private key, only bob has: s_sa = p_spend + H(ss) (1 part) 
+Both the p_send and H(ss) are 256 bits, thus the whole private key is 256 bits. 
+s_sa can decrypt the S_sa for the following reason:
+
+S_sa = s_sa * G (according to EC key generation policy)
+h(ss) * G + P_spend = h(ss) * G + p_spend G = (h(ss) +p_spend)  G 
+
+2. FHE-DKSAP with Paillier
+To use Paillier, we should use the library: from the import paillier and after that, the key processes include three steps:
+- Bob secret key generation and encryption
+- Alice secret key generation and encryption
+- Bob's final key recovery and stealth address generation
+
+## Roadmap
+(What is your proposed timeline? Outline parts of the project and insight on how much time it will take to execute them.)
+- Finalize the proposal of the project (Done)
+- Publish the early study to Ethereum research (Done)
+- Implement the proposal (ing)
+- Complete the benchmark (ing)
+- Finalize the research proposal with benchmark evaluation
+- Revise/propose EIP
+- Implement use cases
+
+## Possible challenges
+(What are the limitations and issues you may need to overcome?)
+
+The communication cost, computation cost and the complexity of implementing fully homomorphic encryption. Meanwhile, most FHE schemes can only support 64 bites. However, our stealth address is 256 bits. The problem lies in how to extend the length for FHE.
+
+## Goal of the project
+(What does success look like? Describe the end goal of the project, scope, state and impact for the project to be considered finished and successful.)
+
+- It protects privacy of stealth addresses by computing over ciphertext. 
+- Compared to DKSAP and BasedSAP, our design removes the risk of leakages on keys and personal information. 
+- Meanwhile, it can prevent quantum computing attacks as well.
```

Comment:
I would love to see more here about what your personal success metrics look like, both for EPF and generally, and less about the actual benefit that FHE-DKSAP will have on the protocol, which you outline in the Motivation section. In what stage does this project need to be for you to consider it to be successful within EPF?